test/integration/message_test.rb

```diff
@@ -32,6 +32,28 @@ def test_edit_message
     assert_no_content msg
   end
 
+  def test_delete_message
+    text = "Here is my Message"
+    blue = users(:blue)
+    red = users(:red)
+    login blue
+    new_msg = blue.send_message_to!(red, text, nil)
+    msg_id = "#private_post_#{new_msg.id}"
+
+    visit "/me/messages/#{red.login}/posts"
+
+    assert_selector msg_id
+
+    hover_and_edit(text) do
+      click_on 'Delete'
+      wait_for_ajax
```

Comment:
This may not be necessary. The assert_no_selector will already wait until the msg_id element disappeared - which is after the ajax returned.
wait_for_ajax is meant to be only used in situations where there are ajax requests that do not affect the dom. For example when filling in a autocompleted field without using the autocomplete. The autocomplete requests may still be on their way even though the form was already submitted (and even it's response rendered).